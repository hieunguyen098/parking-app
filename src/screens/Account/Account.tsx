import { SafeAreaView, StyleSheet, Text, View } from 'react-native';
import React from 'react';
import AccountInfo from './components/AccountInfo';
import MenuAccount from './components/MenuAccount';

const Account = () => {
    return (
        <SafeAreaView style={styles.container}>
            <AccountInfo />
            <MenuAccount />
        </SafeAreaView>
    );
};

export default Account;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
    },
});
