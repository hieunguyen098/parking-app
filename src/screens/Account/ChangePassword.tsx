import { StyleSheet, Text, View } from 'react-native';
import React, { useState } from 'react';
import PasswordInput from '../../components/PasswordInput/PasswordInput';
import BottomButton from '../../components/Buttons/BottomButton';

const ChangePassword = () => {
    const [pin, setPin] = useState('');
    const [newPin, setNewPin] = useState('');
    const [newPinConfirm, setNewPinConfirm] = useState('');
    const [isWrongPassword, setIsWrongPassword] = useState(false);
    const [isNotSame, setIsNotSame] = useState(false);
    const onChange = () => {
        setIsWrongPassword(false);
        setIsNotSame(false);
    };
    const changePassword = () => {
        if (newPin === newPinConfirm) {
            console.log('change passs');
        } else {
            setIsNotSame(true);
        }
    };
    return (
        <>
            <View style={styles.container}>
                <Text style={styles.text}>Nhập mật khẩu hiện tại</Text>
                <PasswordInput
                    warning={{ show: isWrongPassword, message: 'Sai mật khẩu hiện tại' }}
                    length={6}
                    onChange={onChange}
                    value={pin}
                    setValue={setPin}
                />
                <Text style={styles.text}>Nhập mật khẩu mới</Text>
                <PasswordInput
                    warning={{ show: isNotSame, message: 'Mật khẩu mới không khớp' }}
                    length={6}
                    onChange={onChange}
                    value={newPin}
                    setValue={setNewPin}
                />
                <Text style={styles.text}>Nhập lại mật khẩu mới</Text>
                <PasswordInput
                    warning={{ show: isNotSame, message: 'Mật khẩu mới không khớp' }}
                    length={6}
                    onChange={onChange}
                    value={newPinConfirm}
                    setValue={setNewPinConfirm}
                />
            </View>
            <BottomButton title="Xác nhận" onPress={changePassword} />
        </>
    );
};

export default ChangePassword;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        padding: 20,
    },
    text: {
        fontSize: 18,
        textAlign: 'center',
    },
});
