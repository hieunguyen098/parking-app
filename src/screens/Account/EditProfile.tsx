import { StyleSheet, View } from 'react-native';
import React, { useState } from 'react';
import FieldInput from '../../components/FieldInput';
import DateInput from '../../components/DateInput';
import Select from '../../components/Select';
import BottomButton from '../../components/Buttons/BottomButton';
import { useQuery, useQueryClient } from 'react-query';
import { useFocusEffect } from '@react-navigation/native';
import { getUser } from '../../services/user.api';

const genders = [
    {
        id: 0,
        label: 'Nam',
        value: 1,
    },
    {
        id: 1,
        label: 'Nữ',
        value: 0,
    },
    {
        id: 2,
        label: 'Khác',
        value: -1,
    },
];

const EditProfile = () => {
    const [fullname, setFullname] = useState('');
    const [birthday, setBirthday] = useState('');
    const [email, setEmail] = useState('');
    const [gender, setGender] = useState(-1);
    const queryClient = useQueryClient();
    const {
        data: userInfo,
        isLoading,
        error,
    } = useQuery({
        queryKey: ['userInfo'],
        queryFn: () => {
            return getUser();
        },
        retry: false,
        select: (data) => (data.data ? data.data[0] : null),
        onSuccess: (data) => {
            setFullname(data.name);
            setBirthday(data.birthDay);
            setEmail(data.email);
            setGender(data.gender);
        },
    });
    useFocusEffect(
        React.useCallback(() => {
            queryClient.invalidateQueries('userInfo');
        }, []),
    );

    const updateProfile = () => {
        console.log(birthday);
    };
    return (
        <>
            <View style={styles.container}>
                <FieldInput
                    value={fullname}
                    setValue={setFullname}
                    placeHolder="Họ tên"
                    iconPosition="none"
                    keyboardType="text"
                />
                <DateInput
                    value={birthday}
                    setValue={setBirthday}
                    placeHolder="Ngày sinh"
                    source={require('../../../assets/images/calendar-icon.png')}
                    iconPosition="right"
                    keyboardType="text"
                />
                <FieldInput
                    value={email}
                    setValue={setEmail}
                    placeHolder="Email"
                    source={require('../../../assets/images/mail-icon.png')}
                    iconPosition="right"
                    keyboardType="email-address"
                />
                <Select value={gender} setValue={setGender} data={genders} />
            </View>
            <BottomButton title="Cập nhật" onPress={updateProfile} />
        </>
    );
};

export default EditProfile;

const styles = StyleSheet.create({
    container: {
        padding: 20,
        backgroundColor: '#fff',
        flex: 1,
    },
});
