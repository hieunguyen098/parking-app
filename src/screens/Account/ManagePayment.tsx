import { StyleSheet, Text, View, Alert } from 'react-native';
import React from 'react';
import PaymentMethod from './components/PaymentMethod';
import { onAlert } from '../../utils/ui';

const ManagePayment = () => {
    const onDelete = () => {
        console.log('deleted');
    };
    return (
        <View style={styles.container}>
            <Text style={styles.title}>Hình thức thanh toán khả dụng</Text>
            <PaymentMethod
                onDelete={() => onAlert('Xóa liên kết', 'Bạn có chắc chắn muốn xóa?', onDelete)}
                name={'MoMo'}
                source={require('../../../assets/images/momo-logo.png')}
            />
            <Text style={styles.title}>Thêm hình thức thanh toán</Text>
            <View style={styles.listMethod}>
                <PaymentMethod
                    disabled={true}
                    name={'MoMo'}
                    source={require('../../../assets/images/momo-logo-disabled.png')}
                />
                <PaymentMethod
                    disabled={true}
                    name={'ZaloPay'}
                    source={require('../../../assets/images/zalopay-logo-disabled.png')}
                />
            </View>
        </View>
    );
};

export default ManagePayment;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        padding: 20,
    },
    title: {
        fontSize: 14,
        textTransform: 'uppercase',
        marginBottom: 10,
    },
    listMethod: {
        flexWrap: 'wrap',
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
});
