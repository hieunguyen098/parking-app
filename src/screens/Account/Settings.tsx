import { StyleSheet, Text, View, Pressable, Switch } from 'react-native';
import React, { useState } from 'react';
import { GlobalStyles } from '../../constants/style';
import { useNavigation } from '@react-navigation/native';

const Settings = () => {
    const navigation: any = useNavigation();
    const [isEnabledFingerprint, setIsEnabledFingerprint] = useState(false);
    const [isEnabledFace, setIsEnabledFace] = useState(false);
    const toggleSwitchFingerprint = () => setIsEnabledFingerprint((previousState) => !previousState);
    const toggleSwitchFace = () => setIsEnabledFace((previousState) => !previousState);
    return (
        <View style={styles.container}>
            <View style={styles.listButton}>
                <Pressable
                    onPress={() => navigation.navigate('ChangePassword')}
                    style={styles.button}
                    android_ripple={{ color: GlobalStyles.colors.lightGrey100 }}
                >
                    <Text style={styles.text}>Đổi mật khẩu</Text>
                </Pressable>
                <View style={styles.toggleButton}>
                    <Text style={styles.text}>Xác thực vân tay</Text>
                    <Switch
                        trackColor={{
                            false: GlobalStyles.colors.lightGrey100,
                            true: GlobalStyles.colors.primaryOrange,
                        }}
                        style={{ padding: 0 }}
                        thumbColor={isEnabledFingerprint ? '#fff' : '#f4f3f4'}
                        ios_backgroundColor="#3e3e3e"
                        onValueChange={toggleSwitchFingerprint}
                        value={isEnabledFingerprint}
                    />
                </View>
                <View style={styles.toggleButton}>
                    <Text style={styles.text}>Xác thực khuôn mặt</Text>
                    <Switch
                        trackColor={{
                            false: GlobalStyles.colors.lightGrey100,
                            true: GlobalStyles.colors.primaryOrange,
                        }}
                        thumbColor={isEnabledFace ? '#fff' : '#f4f3f4'}
                        ios_backgroundColor="#3e3e3e"
                        onValueChange={toggleSwitchFace}
                        value={!isEnabledFace}
                    />
                </View>
            </View>
        </View>
    );
};

export default Settings;

const styles = StyleSheet.create({
    container: {
        paddingVertical: 20,
    },
    listButton: {
        backgroundColor: '#fff',
    },
    button: {
        paddingHorizontal: 20,
        paddingVertical: 15,
        fontSize: 18,
    },
    text: {
        fontSize: 18,
    },
    toggleButton: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingHorizontal: 20,
        paddingVertical: 8,
    },
});
