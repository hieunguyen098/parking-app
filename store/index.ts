import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './slices/authSlice';
import { generalReducer } from './slices/generalSlice';

export const store = configureStore({
    reducer: {
        auth: authReducer,
        general: generalReducer,
    },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
