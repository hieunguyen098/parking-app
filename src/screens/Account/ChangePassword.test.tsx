import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { store } from '../../../store';
import Account from './Account';
import EditProfile from './EditProfile';
import ChangePassword from './ChangePassword';

jest.useFakeTimers();
test('test change password', () => {
    const queryClient = new QueryClient();
    const component = (
        <QueryClientProvider client={queryClient}>
            <Provider store={store}>
                <NavigationContainer>
                    <ChangePassword />
                </NavigationContainer>
            </Provider>
        </QueryClientProvider>
    );

    const tree = render(component);
    expect(tree).toMatchSnapshot();
});
