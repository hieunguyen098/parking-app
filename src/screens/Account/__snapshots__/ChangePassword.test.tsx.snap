// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test change password 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Nhập mật khẩu hiện tại
    </Text>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          Array [
            Object {
              "backgroundColor": "#FBFBFB",
              "borderColor": "#FBFBFB",
              "borderRadius": 20,
              "borderWidth": 1,
              "justifyContent": "center",
              "marginVertical": 10,
              "paddingHorizontal": 20,
              "paddingVertical": 30,
            },
            false,
            false,
          ],
        ]
      }
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <TextInput
        autoCapitalize="characters"
        autoCorrect={false}
        blurOnSubmit={false}
        caretHidden={true}
        clearButtonMode="never"
        disableFullscreenUI={true}
        keyboardType="number-pad"
        maxLength={6}
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onPressIn={[Function]}
        spellCheck={false}
        style={
          Object {
            "bottom": 0,
            "fontSize": 1,
            "left": 0,
            "opacity": 0.01,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      style={
        Object {
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Nhập mật khẩu mới
    </Text>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          Array [
            Object {
              "backgroundColor": "#FBFBFB",
              "borderColor": "#FBFBFB",
              "borderRadius": 20,
              "borderWidth": 1,
              "justifyContent": "center",
              "marginVertical": 10,
              "paddingHorizontal": 20,
              "paddingVertical": 30,
            },
            false,
            false,
          ],
        ]
      }
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <TextInput
        autoCapitalize="characters"
        autoCorrect={false}
        blurOnSubmit={false}
        caretHidden={true}
        clearButtonMode="never"
        disableFullscreenUI={true}
        keyboardType="number-pad"
        maxLength={6}
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onPressIn={[Function]}
        spellCheck={false}
        style={
          Object {
            "bottom": 0,
            "fontSize": 1,
            "left": 0,
            "opacity": 0.01,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      style={
        Object {
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Nhập lại mật khẩu mới
    </Text>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          Array [
            Object {
              "backgroundColor": "#FBFBFB",
              "borderColor": "#FBFBFB",
              "borderRadius": 20,
              "borderWidth": 1,
              "justifyContent": "center",
              "marginVertical": 10,
              "paddingHorizontal": 20,
              "paddingVertical": 30,
            },
            false,
            false,
          ],
        ]
      }
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <TextInput
        autoCapitalize="characters"
        autoCorrect={false}
        blurOnSubmit={false}
        caretHidden={true}
        clearButtonMode="never"
        disableFullscreenUI={true}
        keyboardType="number-pad"
        maxLength={6}
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onPressIn={[Function]}
        spellCheck={false}
        style={
          Object {
            "bottom": 0,
            "fontSize": 1,
            "left": 0,
            "opacity": 0.01,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>,
  <View
    style={
      Object {
        "borderTopLeftRadius": 24,
        "borderTopRightRadius": 24,
        "bottom": 0,
        "elevation": 1,
        "padding": 20,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FF953A",
            "borderRadius": 8,
            "paddingHorizontal": 12,
            "paddingVertical": 18,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#FF953A",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 18,
              "fontWeight": "600",
              "textAlign": "center",
            },
            Object {
              "color": "#fff",
            },
            undefined,
          ]
        }
      >
        Xác nhận
      </Text>
    </View>
  </View>,
]
`;
